test/clients.go

```diff
@@ -46,21 +46,74 @@ import (
 	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/typed/pipeline/v1beta1"
 	resourceversioned "github.com/tektoncd/pipeline/pkg/client/resource/clientset/versioned"
 	resourcev1alpha1 "github.com/tektoncd/pipeline/pkg/client/resource/clientset/versioned/typed/resource/v1alpha1"
+	rbacv1 "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	"k8s.io/client-go/tools/clientcmd"
 	knativetest "knative.dev/pkg/test"
 )
 
 // clients holds instances of interfaces for making requests to the Pipeline controllers.
 type clients struct {
 	KubeClient *knativetest.KubeClient
 
+	RbacV1Client *rbacv1.RbacV1Client
+
 	PipelineClient         v1beta1.PipelineInterface
+	ClusterTaskClient      v1beta1.ClusterTaskInterface
 	TaskClient             v1beta1.TaskInterface
 	TaskRunClient          v1beta1.TaskRunInterface
 	PipelineRunClient      v1beta1.PipelineRunInterface
 	PipelineResourceClient resourcev1alpha1.PipelineResourceInterface
 	ConditionClient        v1alpha1.ConditionInterface
 }
 
+// newClients instantiates and returns several clientsets required for making requests to the
+// Pipeline cluster specified by the combination of clusterName and configPath. Clients can
+// make requests within namespace.
+func newClientsWithOverrides(t *testing.T, configPath, clusterName, namespace string, overrides *clientcmd.ConfigOverrides) *clients {
+	t.Helper()
+	var err error
+	c := &clients{}
+
+	c.KubeClient, err = knativetest.NewKubeClient(configPath, clusterName)
+	if err != nil {
+		t.Fatalf("failed to create kubeclient from config file at %s: %s", configPath, err)
+	}
+
+	// From BuildClientConfig in knative.dev/pkg/test/clients.go
```

Comment:
not sure what this comment means, could you explain a bit more?

Reply:
My mistake, I didn't mean to include this. I was trying to keep note of where I got snippets from.